src/main/java/org/folio/qm/service/impl/DataImportEventProcessingServiceImpl.java

```diff
@@ -40,19 +41,23 @@ public void processDICompleted(DataImportEventPayload data) {
             updateBuilder.status(RecordCreationStatusEnum.ERROR).errorMessage(errorMessage);
           }
         );
+      result = true;
     } catch (IllegalStateException e) {
       updateBuilder.status(RecordCreationStatusEnum.ERROR).errorMessage(e.getMessage());
     }
     processDIEvent(data, updateBuilder.build());
+    return result;
   }
 
   @Override
-  public void processDIError(DataImportEventPayload data) {
+  public boolean processDIError(DataImportEventPayload data) {
     var errorMessage = extractErrorMessage(data).orElse(ERROR_MISSED_MESSAGE);
     var updateBuilder = RecordCreationStatusUpdate.builder()
       .status(RecordCreationStatusEnum.ERROR)
       .errorMessage(errorMessage);
     processDIEvent(data, updateBuilder.build());
+
+    return true;
```

Comment:
Do we return true in all scenarios?
Lets return result `IsUpdated` of private method `processDIEvent()`.